File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length()!=t.length()){
            return false;
        }
        Map<Character,Integer>mp1=new HashMap<>();
        Map<Character,Integer>mp2=new HashMap<>();
        for(char i:s.toCharArray()){
          mp1.put(i,mp1.getOrDefault(i,0)+i);
        }

        for(char i:t.toCharArray()){
            mp2.put(i,mp2.getOrDefault(i,0)+i);
        }
        if(mp1.equals(mp2)){
            return true;
        }
        return false;
        
    }
}

File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.length()!=t.length()){
            return false;
        }
        map<char,int>atlas1;
        map<char,int>atlas2;
        for(char i:s){
            atlas1[i]++;
        }

         for(char i:t){
            atlas2[i]++;
        }

        if(atlas1==atlas2){
            return true;
        }
        return false;

    }
};